src/map/unit.hpp

```diff
@@ -64,6 +64,20 @@ struct unit_data {
 	int32 group_id;
 
 	std::vector<int32> shadow_scar_timer;
+
+	// Functions and struct to calculate and store exact position at a certain tick
+	int16 getx(t_tick tick = gettick());
+	int16 gety(t_tick tick = gettick());
+	void getpos(int16 &x, int16 &y, uint8 &sx, uint8 &sy, t_tick tick = gettick());
+private:
+	void update_pos(t_tick tick);
+	struct s_udPos {
```

Comment:
```suggestion
	struct {
```
If you want to declare it locally/private you do not have to give it a name.
If you do want to give it a name, then please follow the naming standards for example: s_ud_pos